core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.java

```diff
@@ -285,24 +266,58 @@ private void resetAdapter() {
     searchPresenter.getRecentSearches();
   }
 
-  private ArrayAdapter<String> getDefaultAdapter() {
-    return new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1) {
-      @NonNull
-      @Override
-      public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
-        View row;
+  private DefaultAdapter getDefaultAdapter() {
+    return new DefaultAdapter();
+  }
 
-        if (convertView == null) {
-          row = LayoutInflater.from(parent.getContext())
-            .inflate(android.R.layout.simple_list_item_1, null);
-        } else {
-          row = convertView;
-        }
+  public class DefaultAdapter extends RecyclerView.Adapter<DefaultAdapter.ViewHolder> {
+
+    private Context context;
+    private AdapterView.OnItemClickListener onItemClickListener;
+    List<String> searchList;
+
+    @NonNull @Override
+    public DefaultAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+      View view = (View) LayoutInflater.from(parent.getContext())
+        .inflate(android.R.layout.simple_list_item_1, null);
+      ViewHolder holder = new ViewHolder(view);
+      return holder;
+    }
+
+    @Override public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+      holder.txt.setText(
+        Html.fromHtml(getItem(position)).toString());///need to get rid of this error
+    }
 
-        ((TextView) row).setText(Html.fromHtml(getItem(position)));
-        return row;
+    @Override public int getItemCount() {
+      return searchList.size();
+    }
+
+    @NonNull public String getItem(int position) {
+      return searchList.get(position);
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
+      TextView txt;
```

Comment:
`txt` is a bit weak as names go. `recentSearch` or something like that would be better